client/core/core.go

```diff
@@ -2117,6 +2128,8 @@ func (c *Core) prepareTrackedTrade(dc *dexConnection, form *TradeForm, crypter e
 	result := new(msgjson.OrderResult)
 	err = dc.signAndRequest(msgOrder, route, result, DefaultResponseTimeout)
 	if err != nil {
+		// Do NOT unlock the coins because the request may have actually reached
+		// the server.
 		return nil, 0, fmt.Errorf("new order request with DEX server %v failed: %w", dc.acct.host, err)
```

Comment:
I'm reconsidering this. The client does not save the order to db, nor will it respond to preimage requests for the order. Only advantage to keeping the coins locked is to prevent new orders from using these coins as they're already locked by the server. But we can't keep these locked forever. And there's no client-restart-agnostic reliable way of knowing that the server has also ditched the order, so we can safely unlock coins.

Any objections to unlocking right away?

Reply:
This is an odd case.  It's hard to imagine the request failing but it actually being booked, but if the client cannot do anything with the order, I don't see a reason to keep the coins locked.